source/ht_file.cpp

```diff
@@ -206,14 +182,16 @@ namespace Hatchit
         **/
         size_t File::Read(BYTE* out, size_t len)
         {
-            size_t _len = 0;
+            m_handle.read(reinterpret_cast<char*>(out), len);
+            size_t count = m_handle.gcount();
 
-            _len = fread(out, sizeof(BYTE), len, m_handle);
-
-            if (_len == 0)
-                throw FileException(m_path, errno);
+            //if (count == 0)
```

Comment:
There isn't an `errno` value for the end of a file being reached. What should I do for re-adding in that functionality? Throwing the exception when the count is 0 doesn't work for ASCII files because with the INI reader, empty lines return a length of 0 so I'll edit the documentation for that part.